extensions/cornerstone/src/ViewportOverlay.js

```diff
@@ -1,95 +1,57 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import cornerstone from 'cornerstone-core';
-import classnames from 'classnames';
-
-const ViewportOverlay = ({
-  imageId,
-  scale,
-  windowWidth,
-  windowCenter,
-  imageIndex,
-  stackSize,
-  activeTools,
-}) => {
-  const topLeft = 'top-viewport left-viewport';
-  const topRight = 'top-viewport right-viewport-scrollbar';
-  const bottomRight = 'bottom-viewport right-viewport-scrollbar';
-  const bottomLeft = 'bottom-viewport left-viewport';
-  const overlay = 'absolute pointer-events-none';
-
-  const isZoomActive = activeTools.includes('Zoom');
-  const isWwwcActive = activeTools.includes('Wwwc');
-
-  if (!imageId) {
-    return null;
+import classnames from 'classnames'
+import ConfigPoint from 'config-point'
```

Comment:
Yes, thats what this code does as it is written.  It also allows runtime configurable extensions to it, for example, by adding the ?theme=viewportOverlay parameter one could load additional overlay extensions.   Extensions don't have to depend on config-point to generate overlays - there are multiple options here:
1. Inherit an extension of the default overlay, and extend it with specific extensions using config-point
2. Use the base default overlay, without any extensions by just using the default component (no config-point involved in the extension side of this)
3. Write a custom overlay, without config point involved at all, just using it directly.
4. Write an extension that modifies another (earlier) extension's overlay by using config-point to customize the earlier extensions view.

Neither of 2 or 3 depend on config-point except indirectly.  Again, what I'm trying to do is decouple things that sites or extensions may want to configure from the extensions themselves in a way that doesn't require re-writing extensions or inheritting from an extension in a way that tightly couples the change into the actual implementation of the extension.

Reply:
I see, thanks for this. The URL params is an interesting one, we can pass this to the mode, but it would be a bit clunky to somehow pass this to extensions, although possible.

> Again, what I'm trying to do is decouple things that sites or extensions may want to configure from the extensions themselves in a way that doesn't require re-writing extensions or inheritting from an extension in a way that tightly couples the change into the actual implementation of the extension.

I 100% get the motivation, where I'm wondering if we can improve is mostly on the formalization of this. The configuration being centrally managed and passed down to modes and extensions might keep things more logically seperated. Right now its unclear at what layer config-point should be used, as it looks like the configuration can be defined on any level. I may be wrong.

P.S. I think the problem you're trying to solve might be solvable (at least partially) by us thinking harder about how flexible the UI components in @ohift/ui themselves are, making the data flow of the configuration required by an extension/mode/application more unidirectional. Happy to discuss this tomorrow.